DeleteUserDataActivity/backupAndDelete.ts

```diff
@@ -307,38 +326,49 @@ const backupAndDeleteAllNotificationsData = ({
   readonly notificationModel: NotificationDeletableModel;
   readonly notificationStatusModel: NotificationStatusDeletableModel;
   readonly userDataBackup: IBlobServiceInfo;
-}): TaskEither<QueryFailure, true> =>
-  notificationModel.findNotificationForMessage(message.id).foldTaskEither(
-    failure =>
+}): TE.TaskEither<QueryFailure, true> =>
+  pipe(
+    notificationModel.findNotificationForMessage(message.id),
+    TE.fold(
       // There are cases in which a message has no notification.
       // We just consider the notification to be deleted
-      failure.kind === "COSMOS_ERROR_RESPONSE" && failure.error.code === 404
-        ? taskEither.of(true)
-        : fromLeft(toQueryFailure(failure)),
-    maybeNotification =>
-      maybeNotification.fold(
-        // There are cases in which a message has no notification.
-        // We just consider the notification to be deleted
-        taskEither.of(true),
-        // For the found notification, we delete its statuses before deleting the notification itself
-        notification =>
-          backupAndDeleteNotificationStatus({
-            notification,
-            notificationStatusModel,
-            userDataBackup
-          })
-            .chain(() =>
-              backupAndDeleteNotification({
-                notification,
-                notificationModel,
-                userDataBackup
-              })
-            )
-            .bimap(
-              e => toQueryFailure(new Error(e.reason)),
-              () => true
-            )
-      )
+      failure =>
+        failure.kind === "COSMOS_ERROR_RESPONSE" && failure.error.code === 404
+          ? TE.of(true)
+          : TE.left(toQueryFailure(failure)),
+
+      maybeNotification =>
```

Comment:
We ca use `flow` here.

Reply:
fixed in https://github.com/pagopa/io-functions-admin/pull/165/commits/67fd68c4a886e99b4273edbada5de7f5600e40b8